debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsSnapshotProducer.java

```diff
@@ -203,6 +203,7 @@ private void takeSnapshot(BlockingConsumer<ChangeEvent> consumer) {
                 // process and send the last record after marking it as such
                 logger.info("Step 5: sending the last snapshot record");
                 sourceInfo.markLastSnapshotRecord();
+                changeSourceToLastSnapshotRecord(currentRecord);
```

Comment:
Hum, something is weird here: if I comment out this line, still all tests that make use of `assertRecordOffsetAndSnapshotSource()` pass. Shouldn't any of them fail? Also, why isn't this part of `markLastSnapshotRecord()` in the line above?

Reply:
@gunnarmorling Thanks for spotting, I inadverently removed one assert when I was copying around the code to handle heartbeat events.

Reply:
Ok, got it. It feels a bit convoluted, but I can't think of a really better way right now either.